floor_generator/lib/misc/extension/dart_object_extension.dart

```diff
@@ -23,12 +31,20 @@ extension DartObjectExtension on DartObject {
   /// get the ForeignKeyAction this enum represents,
   /// or the result of `null` if the enum did not contain a valid value
   ForeignKeyAction? toForeignKeyAction() {
-    final enumValueString = toEnumValueString();
-    if (enumValueString == null) {
-      return null;
-    } else {
-      return ForeignKeyAction.values.singleWhereOrNull(
-          (foreignKeyAction) => foreignKeyAction.toString() == enumValueString);
+    final interfaceType = type as InterfaceType;
+    final enumFields = interfaceType.element.fields
+        .where((element) => element.isEnumConstant)
+        .toList();
+
+    // Find the index of the matching enum constant.
+    ForeignKeyAction? enumValue;
+    for (int i = 0; i < enumFields.length; i++) {
+      if (enumFields[i].computeConstantValue() == this) {
+        enumValue = ForeignKeyAction.values[i];
+        break;
+      }
```

Comment:
Here it is the same as described above.
With `_getIndex` you can replace it with
```
    final enumIndex = _getIndex(enumFields);
    final enumValue = enumIndex != null
            ? ForeignKeyAction.values[enumIndex]
            : null;
```